x/concentrated-liquidity/fees_test.go

```diff
@@ -699,3 +699,114 @@ func (s *KeeperTestSuite) TestCollectFees() {
 		})
 	}
 }
+
+func (s *KeeperTestSuite) TestUpdateFeeAccumulatorPosition() {
+	var (
+		ownerOne = s.TestAccs[0]
+		ownerTwo = s.TestAccs[1]
+	)
+
+	type updateFeeAccumPositionTest struct {
+		poolId        uint64
+		owner         sdk.AccAddress
+		liquidity     sdk.Dec
+		lowerTick     int64
+		upperTick     int64
+		expectedError error
+	}
+
+	positions := map[int]updateFeeAccumPositionTest{
+		1: {
+			lowerTick: DefaultLowerTick,
+			upperTick: DefaultUpperTick,
+		},
+		2: {
+			lowerTick: DefaultLowerTick + 1,
+			upperTick: DefaultUpperTick,
+		},
+		3: {
+			lowerTick: DefaultLowerTick,
+			upperTick: DefaultUpperTick + 1,
+		},
+	}
+
+	tests := map[string]updateFeeAccumPositionTest{
+		"update position with a different pool ID": {
+			poolId:    2,
+			owner:     ownerOne,
+			liquidity: DefaultLiquidityAmt,
+			lowerTick: DefaultLowerTick,
+			upperTick: DefaultUpperTick,
+		},
+		"update position with a different owner": {
+			poolId:    1,
+			owner:     ownerTwo,
+			liquidity: DefaultLiquidityAmt,
+			lowerTick: DefaultLowerTick,
+			upperTick: DefaultUpperTick,
+		},
+		"update position with a different lower tick": {
+			poolId:    1,
+			owner:     ownerOne,
+			liquidity: DefaultLiquidityAmt,
+			lowerTick: DefaultLowerTick + 1,
+			upperTick: DefaultUpperTick,
+		},
+		"update position with a different upper tick": {
```

Comment:
It should be possible to test error cases too. Can we cover cases such as pool does not exist, fee accumulator position is not pre-initialized.

I think the latter is very important to cover